providers/task.rb

```diff
@@ -36,12 +36,22 @@
     validate_create_day
     validate_create_months
 
-    schedule = @new_resource.frequency == :on_logon ? 'ONLOGON' : @new_resource.frequency
+    schedule = case @new_resource.frequency
+               when :on_logon
+                 'ONLOGON'
+               when :on_idle
+                 'ONIDLE'
+               else
+                 @new_resource.frequency
+               end
     frequency_modifier_allowed = [:minute, :hourly, :daily, :weekly, :monthly]
     options = {}
     options['F'] = '' if @new_resource.force || task_need_update?
     options['SC'] = schedule
-    options['MO'] = @new_resource.frequency_modifier if frequency_modifier_allowed.include?(@new_resource.frequency)
+    unless @new_resource.frequency == :monthly && !%w(FIRST SECOND THIRD FOURTH LAST LASTDAY).include?(@new_resource.frequency_modifier)
```

Comment:
Just realized it's actually /M and /MO that can't go together so I'll simplify this and it will still fit in one line as it did originally (just adding "&& @new_resource.months.nil?" to it)
This is from a manual test I did with schtasks.exe:
ERROR: /M cannot be specified with /MO.
Type "SCHTASKS /CREATE /?" for usage.